### PR TITLE
basic setup for the webinar details portal

### PR DESCRIPTION
**Pull request by subhrapaladhi**
### Things done: 
- creating sessions and authentication of the login
- adding new webinar data to database
- setting up endpoint to send list of all webinars(and their details) from database as a json response

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+
+// setting the home directory
+const publicDir = require('path').join(__dirname,'/public');
+app.use(express.static(publicDir));
+
+// setup for creating session and other related stuff with authentication
+app.use(session({
+    secret: 'fldsk2232',
```
Comment (L04DB4L4NC3R) [inline app.js]: Don't leave the secrets in paintext. Hide them in an environment file

File: app.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+let auth = function(req,res,next){
+    if(req.session && req.session.password == password)
+        return next();      // if a valid session is already running then continue 
+    
+    else
+        return res.redirect('/');       // otherwise redirect to login form page
+}
+
+const password = "abc123";
```
Comment (L04DB4L4NC3R) [inline app.js]: Don't save the password in plaintext. Hide it in an environment file

File: app.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    
+    else
+        return res.redirect('/');       // otherwise redirect to login form page
+}
+
+const password = "abc123"; 
+
+
+// setting up db
+mongoose.Promise = global.Promise;
```
Comment (L04DB4L4NC3R) [inline app.js]: No need for this line

File: app.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+mongoose.connect('mongodb://localhost/webinar', {useNewUrlParser: true});
```
Comment (L04DB4L4NC3R) [inline app.js]: Don't save the mongoDB url in plaintext. Hide it in an environment file

File: app.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    res.render('login.ejs')
+})
+
+app.post('/', (req,res) => {
+    console.log(req.session);
+    let enteredPassword = req.body.password;
+    console.log(enteredPassword)
+    if(password == enteredPassword){
+        console.log('logged in');
+        req.session.password = enteredPassword;
```
Comment (L04DB4L4NC3R) [inline app.js]: Don't save the password in the session. Save only the user ID. Or in this case, a boolean saying logged in or not

File: app.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    new Webinar(newWebinar).save()
+                            .then((temp) =>console.log(temp))
+                            .catch((err) => console.log(err))
+
+    res.send('home/newWebinar-post')
+})
+
+app.get('/logout', auth, (req,res) => {
+    req.session.destroy();
+    res.send('logout success');
```
Comment (L04DB4L4NC3R) [inline app.js]: redirect to the login page after logout

**Comment by subhrapaladhi** [comment]
@L04DB4L4NC3R 
i will do required ASAP

**Review by subhrapaladhi — COMMENTED** [review]
I have used a npm library named env2 for hiding secret, password and mongodb url as environment variable in the .env file and featching them then to app.js. I have added the .env file to .gitignore. [Link for the git repo of env2](https://github.com/dwyl/env2)

I am also using boolean instead of saving the password in the session and redirecting to login page after logout.

**Pull request merged**